lib/Doctrine/Enumeration/Enumerated.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Enumeration;
+
+use Doctrine\Enumeration\Exception\NotSupportedException;
+use function array_search;
+use function constant;
+use function in_array;
+
+trait Enumerated
```

Comment:
Please mark this as `@internal` - we don't want userland to know/use this

Reply:
Maybe have a look at https://github.com/marc-mabe/php-enum.